server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -210,7 +210,8 @@ public Response renameSecret(
         .orElseThrow(NotFoundException::new);
     secretDAO.renameSecretById(secret.series().id(), newName, automationClient.getName());
 
-    return Response.ok().build();
+    UriBuilder uriBuilder = UriBuilder.fromResource(SecretResource.class).path(newName);
+    return Response.created(uriBuilder.build()).build();
```

Comment:
Rather than just saying "ok", we indicate that a new resource (the renamed secret) was created and return the URL to that resource.